lib/mastercard/oauth.rb

```diff
@@ -239,17 +240,16 @@ def encode_uri_component(uri_component)
       #
       def get_nonce(len = 32)
         # Returns a random string of length=len
-        o = [('a'..'z'), ('A'..'Z'), (0..9)].map(&:to_a).flatten
-        (0...len).map { o[rand(o.length)] }.join
+        SecureRandom.alphanumeric(len)
```

Comment:
Unfortunately we cannot use SecureRandom as older versions of ruby don't support it and we don't want to break backwards compatibility. See here for the broken tests: https://github.com/Mastercard/oauth1-signer-ruby/actions/runs/6096972234/job/16543657471

Reply:
After some discussion internally and research, we agree with your point. We will consider version 2.4 unsupported going forward.
Thanks again @amasses !

Reply:
Released version 1.1.3: https://rubygems.org/gems/mastercard_oauth1_signer/versions/1.1.3

Reply:
Thanks @danny-gallagher , much appreciated!